api-registration/server/index.js

```diff
@@ -0,0 +1,73 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+        insert into "users" ("username", "hashedPassword")
+        values ($1, $2)
+        returning *
+      `;
+      const values = [username, hashedPassword];
+      db.query(sql, values)
+        .then(result => {
+          res.status(201).json({
+            userId: result.rows[0].userId,
+            username: username,
+            createdAt: result.rows[0].createdAt
+          });
```

Comment:
This works but it is easier to update your SQL statements `returning` clause to just return the columns you want to send to the client, then you can do:
```JAVASCRIPT
res.status(201).json(result.rows[0]);
```